applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -499,7 +500,58 @@ void SmallStrainUPwDiffOrderElement::CalculateOnIntegrationPoints(const Variable
                 std::inner_product(shape_function_values.begin(), shape_function_values.end(),
                                    nodal_hydraulic_head.begin(), 0.0);
         }
-    } else {
+    } else if (rVariable == CONFINED_STIFFNESS || rVariable == SHEAR_STIFFNESS) {
+        size_t variable_index = 0;
+        if (rVariable == CONFINED_STIFFNESS) {
+            if (r_geom.WorkingSpaceDimension() == 2) {
+                variable_index = INDEX_2D_PLANE_STRAIN_XX;
+            } else if (r_geom.WorkingSpaceDimension() == 3) {
+                variable_index = INDEX_3D_XX;
+            } else {
+                KRATOS_ERROR << "CONFINED_STIFFNESS can not be retrieved for dim "
+                             << r_geom.WorkingSpaceDimension() << " in element: " << this->Id()
+                             << std::endl;
+            }
+        } else if (rVariable == SHEAR_STIFFNESS) {
+            if (r_geom.WorkingSpaceDimension() == 2) {
+                variable_index = INDEX_2D_PLANE_STRAIN_XY;
+            } else if (r_geom.WorkingSpaceDimension() == 3) {
+                variable_index = INDEX_3D_XZ;
+            } else {
+                KRATOS_ERROR << "SHEAR_STIFFNESS can not be retrieved for dim "
+                             << r_geom.WorkingSpaceDimension() << " in element: " << this->Id()
+                             << std::endl;
+            }
+        }
+
+        ElementVariables Variables;
+        this->InitializeElementVariables(Variables, rCurrentProcessInfo);
+
+        const auto b_matrices = CalculateBMatrices(Variables.DNu_DXContainer, Variables.NuContainer);
+        const auto deformation_gradients = CalculateDeformationGradients();
+        auto       strain_vectors        = StressStrainUtilities::CalculateStrains(
```

Comment:
This can probably be const:
```suggestion
        const auto strain_vectors        = StressStrainUtilities::CalculateStrains(
```

Reply:
it cant, it can be altered in CalculateAnyOfMaterialResponse